### ISSUE TITLE
App "example_power_gating_core" cannot be built

### ISSUE THREAD

**Issue by JuanSapriza** [esl-epfl/x-heep#261]
During the test for PR #239 it was seen that a parameter number issue was introduced into the `example_power_gating_core` app, preventing it from being built. The problem was existing before, but was hiding in a `#ifdef` that evaluated as `false`, so the code was not compiled. 
This blocks the deployment of the CI of #239
Below the output of the compiler. This was built with GCC on the [X-HEEP builder docker container](https://github.com/orgs/esl-epfl/packages/container/package/xheep-compiler).

```
 /__w/x-heep/x-heep/sw/applications/example_power_gating_core/main.c: In function 'main':
/__w/x-heep/x-heep/sw/applications/example_power_gating_core/main.c:126:5: error: too few arguments to function 'gpio_irq_set_trigger'
  126 |     gpio_irq_set_trigger(&gpio, 1 << 31, kGpioIrqTriggerLevelHigh);
      |     ^~~~~~~~~~~~~~~~~~~~
In file included from /__w/x-heep/x-heep/sw/applications/example_power_gating_core/main.c:16:
/__w/x-heep/x-heep/sw/device/lib/drivers/gpio/gpio.h:224:15: note: declared here
  224 | gpio_result_t gpio_irq_set_trigger(const gpio_t *gpio,
      |               ^~~~~~~~~~~~~~~~~~~~
make[4]: *** [CMakeFiles/main.elf.dir/build.make:66: CMakeFiles/main.elf.dir/applications/example_power_gating_core/main.c.obj] Error 1
```

**Comment by JoseCalero** [comment]
This needs to be fixed by @Ho3eintaji in #246

**Comment by Ho3eintaji** [comment]
it is still previous version, however, pass pin input w/o shift and state is not used so just pass sth, then should be fine.
